clap_mangen/src/render.rs

```diff
@@ -74,18 +72,46 @@ pub(crate) fn synopsis(roff: &mut Roff, cmd: &clap::Command) {
         line.push(roman(" "));
     }
 
-    if cmd.has_subcommands() {
-        let (lhs, rhs) = subcommand_markers(cmd);
-        line.push(roman(lhs));
-        line.push(italic(
-            cmd.get_subcommand_value_name()
-                .unwrap_or_else(|| subcommand_heading(cmd))
-                .to_lowercase(),
-        ));
-        line.push(roman(rhs));
+    line
+}
+
+pub(crate) fn synopsis(roff: &mut Roff, cmd: &clap::Command) {
+    let name = cmd.get_bin_name().unwrap_or_else(|| cmd.get_name());
+    let flatten = cmd.is_flatten_help_set();
+
+    let mut ord_v = Vec::new();
+    if flatten {
+        for subcommand in cmd.get_subcommands() {
+            ord_v.push((
+                subcommand.get_display_order(),
+                format!("{} {}", name, subcommand.get_name().to_owned()),
+                subcommand,
+            ));
+        }
+        ord_v.sort_by(|a, b| (a.0, &a.1).cmp(&(b.0, &b.1)));
+    } else {
+        ord_v.push((cmd.get_display_order(), name.to_string(), cmd));
     }
 
-    roff.text(line);
+    for (_, name, cmd) in ord_v {
+        let mut line = command_with_args(cmd, name);
+
+        if cmd.has_subcommands() && !flatten {
```

Comment:
I'm quite sure you don't love this way of implementing it, do you have a cleaner idea?

Reply:
where possible, we should match how our regular usage renderer does things

Reply:
I changed the style to be less invasive, what do you think?